botocore/httpsession.py

```diff
@@ -177,6 +185,21 @@ def mask_proxy_url(proxy_url):
     return proxy_url
 
 
+def _is_ipaddress(host):
+    """Detects whether the hostname given is an IPv4 or IPv6 address.
+    Also detects IPv6 addresses with Zone IDs. This method is taken from urllib3
+    and uses the regex patterns from urllib3. The only change is we match
+    IPv6 address with braces, as hostname at this point will have square braces around it.
+
+    :param str host: Hostname to examine.
+    :return: True if the hostname is an IP address, False otherwise.
+    """
+    if not six.PY2 and isinstance(host, bytes):
+        # IDN A-label bytes are ASCII compatible.
+        host = host.decode("ascii")
+    return bool(IPV4_RE.match(host) or IPV6_ADDRZ_RE.match(host))
+
+
```

Comment:
```suggestion
    """Wrapper for urllib3's is_ipaddress to provide support for bracketed IPv6 addresses.

    :param str host: Hostname to examine.
    :return: True if the hostname is an IP address, False otherwise.
    """
    return is_ipaddress(host) or bool(IPV6_ADDRZ_RE.match(host))
```

You'll also want to update this import to be `from botocore.utils import IPV6_ADDRZ_RE` for backwards compatibility.

Reply:
Thanks , I've added this now.